src/IssueViz/Editor/LocationTagging/LocationTagger.cs

```diff
@@ -73,25 +74,41 @@ private void OnIssuesChanged(object sender, IssuesChangedEventArgs e)
         private void UpdateTags()
         {
             var textSnapshot = buffer.CurrentSnapshot;
+            var oldTags = TagSpans;
+            TagSpans = CreateTagSpans(textSnapshot);
 
+            var affectedSpan = CalculateSpanOfAllTags(oldTags, textSnapshot);
+            NotifyTagsChanged(affectedSpan);
+        }
+
+        private List<ITagSpan<IIssueLocationTag>> CreateTagSpans(ITextSnapshot textSnapshot)
+        {
             // Get the new locations and calculate the spans if necessary
             var locations = locationService.GetLocations(GetCurrentFilePath());
-            EnsureSpansExist(locations, textSnapshot);
 
-            var oldTags = TagSpans;
-            TagSpans = locations.Select(CreateTagSpan).ToList();
+            EnsureUpdatedSpansExist(locations, textSnapshot);
 
-            var affectedSpan = CalculateSpanOfAllTags(oldTags, textSnapshot);
-            NotifyTagsChanged(affectedSpan);
+            var tagSpans = locations
+                .Where(x => x.Span.HasValue && !x.Span.Value.IsEmpty)
```

Comment:
Repro scenario:
1. Open file with secondary location
2. Edit, save and close the file.
3. Open the file ==> new span will be empty and an exception will be thrown in GetTags TranslateTo